Source/carthage/Archive.swift

```diff
@@ -25,7 +25,15 @@ public struct ArchiveCommand: CommandType {
 					|> flatMap(.Concat) { platform in
 						let framework = platform.relativePath.stringByAppendingPathComponent(options.frameworkName).stringByAppendingPathExtension("framework")!
 						let dSYM = framework.stringByAppendingPathExtension("dSYM")!
-						return SignalProducer(values: [framework, dSYM])
+						if NSFileManager.defaultManager().fileExistsAtPath(framework) {
```

Comment:
It's preferable to invoke `dwarfdump` and catch an error than it is to just check if the input file exists to begin with?

That would also squelch any other legitimate errors arising from `dwarfdump`, which may or may not be preferable.